### PR TITLE
Config Updates

### PR DESCRIPTION
**Pull request by selfish**
### Background and Notable Changes:

- Add .editorconfig
- Update eslint config and ignore
- Update license
- Update dev html to match config
- Move to yarn and upgrade dependencies
- Update ts and rollup configurations
- Update hacs.json
- Add .tool-verions for package manager support
- Add a proper PR template
- Update GH workflows to auto-release on push to main

### CODE REVIEW

**Comment by selfish** [comment]
Hey @avataar,

I noticed you might have seen the new PR with all the config updates and stuff. I'd love to get your thoughts on it! You can check it out and drop any comments or suggestions, pull the branch and tweak things yourself if you like, and of course let me know if you think something could be done differently or better.

I'm totally open to different ways of doing things, so don't hold back!

Just a heads-up, the CI is failing right now because we don't have any releases yet, and maybe due to the repo name issue. But it's not related to the content itself, and everything seems to work on this branch.

When we merge it, a release will be created automatically, and that should fix the CI stuff. But if you have other ideas, feel free to share.

Looking forward to hearing from you!

Cheers!

File: package.json
Comment (ThomDietrich) [inline package.json:15]:
```suggestion
    "url": "git+https://github.com/rejuvenate/lovelace-sun-card.git"
```

File: hacs.json
Comment (ThomDietrich) [inline hacs.json:2]: Agree to stay with the original name. I did add a note about the fork to the repo description!

Would you by any chance know how to add an icon to the project?

image

**Comment by ThomDietrich** [comment]
@avataar I would appreciate your feedback as I have almost no experience with js/ts toolchains.

@selfish your PR is aimed against the "main" branch. Shall we merge this one first, then merge the develop branch?

**Comment by selfish** [comment]
> @avataar I would appreciate your feedback as I have almost no experience with js/ts toolchains.
> 
> @selfish your PR is aimed against the "main" branch. Shall we merge this one first, then merge the develop branch?

I think either order of merging is acceptable, but this PR is quite opinionated and requires a thorough review and input from @avataar. Additionally, @avataar can suggest the best approach for the merge order, as I believe we should integrate his work and fixes from the `develop` branch.

I don't mind reworking this after merging @avataar's work if that's easier.

File: LICENSE
Comment (avataar) [inline LICENSE:3]: I believe we should keep the original copyright and simply add ours after it (also note the project name change).

File: .github/workflows/main.yaml
Comment (avataar) [inline .github/workflows/main.yaml:41]: There's no "test" script to run - is that an error or intended for now?

**Comment by avataar** [comment]
It doesn't build for me because of lint errors:
```
/Users/avataar/tmp/tmp/lovelace-sun-card/src/cardContent.ts
  1:1  error  Run autofix to sort these imports!  simple-import-sort/imports

/Users/avataar/tmp/tmp/lovelace-sun-card/src/index.ts
    1:1   error    Run autofix to sort these imports!  simple-import-sort/imports
  152:65  warning  Forbidden non-null assertion        @typescript-eslint/no-non-null-assertion
  175:22  warning  Forbidden non-null assertion        @typescript-eslint/no-non-null-assertion
  193:26  warning  Forbidden non-null assertion        @typescript-eslint/no-non-null-assertion
  197:9   error    Unexpected console statement        no-console

✖ 6 problems (3 errors, 3 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```

I was able to run eslint --fix and removed the offending console.error() call and then it worked (but still shows the warnings).

Regarding the merge order I'm not sure if you're aware that the develop branch (from the original project, not just my fork) has lots of changes that aren't present in the main branch (including to some of those build files that @selfish changes with this PR). Thus I believe we should ditch the existing main and rename develop to main, then rebase this onto the new main.

**Comment by selfish** [comment]
Makes sense. How about we do this:
You create dev here as you did in your repo.
I'll then rebase this on your develop, fix lint errors, etc. We'll merge this to your dev, see that everything is in order, then merge all to main?

**Comment by selfish** [comment]
Also, sorry for lraving the lint errors in there, I neglected to push fixes for these

File: LICENSE
Comment (ThomDietrich) [reply LICENSE:3]: Good point.

**Comment by ThomDietrich** [comment]
Hey @selfish,
isn't that overly complicated? I am with @avataar on this: Let's just merge develop, then merge your PR. After that @avataar can create a PR for his changes.
Let's take one step at a time.

**Comment by avataar** [comment]
I opened a PR from my develop to this develop - no big changes there so order shouldn't matter.

After we end up with a single up-to-date main branch here I'll open another PR with some other improvements that affect the failing tests.

**Comment by selfish** [comment]
Sure, go ahead!

**Comment by avataar** [comment]
I've closed my original PR that combined my old 3 PRs and opened 3 new PRs instead so it's clearer what/why.

**Comment by ThomDietrich** [comment]
Hey @selfish,
I've merged the develop branch and @avataar 's smaller PRs. We are now ready for your toolchain changes here. Hope resolving conflicts isn't that difficult. Maybe it would be easier for you to start with a fresh branch? Cheers!

**Comment by edwardtfn** [comment]
> It doesn't build for me because of lint errors:

It built for me without any changes (other than the ones related to the name that I just discovered you have done before me). I have the lovelace-sun-card.js in another PR. I think we should prioritise uploading that file as it is required to the hacs validation (which is required to add to HACS default repository list).

File: LICENSE
Comment (edwardtfn) [reply LICENSE:3]: Indeed. I also kept the original copyright when I've setup my fork to publish in HACS.
But there is another file where is @AirtonDB username and email as author. I think we should keep his name, but probably remove his email address as I  belive he archived his repo due to no intention to offer support to it anymore.

File: hacs.json
Comment (edwardtfn) [reply hacs.json:2]: I'm pretty sure icons are not available for lovelace.
If you filter hacs list for lovelace items only there's no single one with an Icon.

**Comment by selfish** [comment]
I'll take care of it soon. I'll make sure everything is resolved, we have a successful build, and we release with a changelog.

**Comment by selfish** [comment]
@ThomDietrich @avataar - ready for review.

**Comment by selfish** [comment]
Note that once this PR is merged, release v0.2.0 will automatically be created, and HACS action will no longer complain about structure.

File: package.json
Comment (avataar) [reply package.json:8]: Yes, I use the dev thing and even made it way more useful (will post PR later).

File: package.json
Comment (avataar) [reply package.json:4]: Let's stick with the shorter description for now. It's presented in a list with a limited width so shorter is better.

File: package.json
Comment (avataar) [inline package.json:25]: Quite a few a of these got downgraded. I assume it was a side effect of starting this off the main branch before develop was merged so let's get the newer versions again (unless there was a legit reason to downgrade).

File: tsconfig.json
Comment (avataar) [inline tsconfig.json:33]: Is there a legit reason something from node_modules is in the include list? It seems to work just fine without it.

**Comment by avataar** [comment]
It's starting to look really good. I resolved most of the suggestions and left some new comments. Here's a summary of what's left to resolve before we can merge this:

- A bunch of downgraded versions in package.json (see comment)
- Many (but not all) .ts files seem to have been relinted with semicolons/no spaces after function names -- it's certainly a matter of debate which style to use but this goes way beyond "config updates" and we lose history. I strongly suggest you revert those. I'm guessing that was unintentional.
- Unexpected include from node_modules in tsconfig.json
- .eslintrc.cjs (new, used now) vs .eslintrc (old, unused) - the old file should be removed and maybe you should review some key differences between the two files. Notably, the old one has rules that would've prevented the relinting of all those .ts files:
```
    "semi": ["error", "never"],
    "space-before-function-paren": ["error", "always"],
```

**Comment by avataar** [comment]
@selfish decided to work on his own fork instead, so I'll fix the remaining issues.

**Comment by ThomDietrich** [comment]
@avataar amazing, yes please do! Thanks! Users are getting nervous and it would be great if we could publish a version asap

**Comment by edwardtfn** [comment]
Anything I could help?

**Comment by avataar** [comment]
Everything should be OK now. If anyone wants to have a final look at the changes I'd appreciate it.

File: package.json
Comment (edwardtfn) [reply package.json:4]: I agree about going with the shorter version, but I think we should fix capitalization:

`"Home Assistant Sun card based on Google Weather design",`

File: rollup.config.dev.mjs
Comment (edwardtfn) [inline rollup.config.dev.mjs:1]: > `file: "./dist/lovelace-sun-card.js",`

Is this correct? On `dev/index.html` you removed the reference to folder `/dist`.

File: rollup.config.mjs
Comment (edwardtfn) [inline rollup.config.mjs:1]: Same comment about `/dist` folder.

File: tsconfig.json
Comment (edwardtfn) [inline tsconfig.json:1]: Are those `,` at the end of lines 28, 32 and 35 needed?

**Review by edwardtfn — APPROVED** [review]
Let's merge so we fix any remaining issues required for building (and test in HACS). Cosmetics can come later.

File: .github/workflows/validate.yaml
Comment (avataar) [reply .github/workflows/validate.yaml:1]: Yes, I saw your PR and will merge it after this one.

Comment (edwardtfn) [reply .github/workflows/validate.yaml:1]: By the way, I don't think that alone will fix the failing validation you have now, as it is failing due to not having the proper structure probably because it is missing the file `lovelace-sun-card.js`.

File: rollup.config.dev.mjs
Comment (avataar) [reply rollup.config.dev.mjs:1]: I think it's just a remnant from this file originally being a copy of the prod rollup config, while the relevant part is the serve section. Running "yarn dev" certainly works as expected.

File: rollup.config.mjs
Comment (avataar) [reply rollup.config.mjs:1]: This one just packages it to dist/lovelace-sun-card.js and works too. Am I missing something?

File: tsconfig.json
Comment (avataar) [reply tsconfig.json:1]: Either way is fine. For me personally having commas allows you to reorder things easily.

File: package.json
Comment (avataar) [reply package.json]: Good point, fixed.

**Review by avataar — APPROVED** [review]

**Pull request merged**
